### PR TITLE
reimplement paperjs functionality after ryan's changes

### PR DESCRIPTION
**Pull request by nayancprakash**
## Description
* reimplements the paper.js functionality after ryan's changes
* allows owner to create new pages

### CODE REVIEW

File: app/javascript/components/layer.jsx [hunk 1]
```diff
@@ -48,165 +47,159 @@ export function Layer({ layer, isOwner, layerId, activeTool, activeColor }) {
...
-
-    // props.layer.draw();
+    let path = null;
+
+    let rect = null;
+    let p1 = null;
+    let p2 = null;
+
+    if (scope) {
+
```
Comment (calebschoepp) [inline app/javascript/components/layer.jsx]: Nit: No newline here

File: app/javascript/components/layer.jsx [hunk 2]
```diff
@@ -48,165 +47,159 @@ export function Layer({ layer, isOwner, layerId, activeTool, activeColor }) {
...
+                p.fullySelected = false;
+              }
+              newPathState.push(p);
+            }
+            setPathState(newPathState);
+          }
+        }
+      };
+
+      scope.view.onKeyDown = (event) => {
```
Comment (calebschoepp) [inline app/javascript/components/layer.jsx:173]: Am I missing something, shouldn't there be a `transmitDiff` called on something like `onKeyUp`?

**Review by calebschoepp — APPROVED** [review]
A nit and a question but otherwise LGTM!

File: app/javascript/components/layer.jsx (hunk 2 above)
Comment (nayancprakash) [reply app/javascript/components/layer.jsx:173]: Yes there are a few places missing `transmitDiff` including text diffs, deletion diffs, selection and moving diffs, and eraser diffs. I only added `transmitDiff` where I did so we could try and tackle the simplest case first, the addition of a pen diff.

**Pull request merged**